### DISCUSSION TITLE
does native control host work on mobile ?

### DISCUSSION THREAD

**Discussion by SaverinOnRails** [AvaloniaUI/Avalonia#11903] [General]
If so, how can we embed a native view like perhaps an android activity or android widget?

**Comment by SaverinOnRails** [comment]
Any update on this?

**Comment by maxkatz6** [comment]
Yes, it works.
You can create an instance of AndroidViewControlHandle from android view and return it from the native control host.
Something like this https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog.Android/EmbedSample.Android.cs

**Comment by SaverinOnRails** [comment]
> Yes, it works.
> You can create an instance of AndroidViewControlHandle from android view and return it from the native control host.
> Something like this https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog.Android/EmbedSample.Android.cs

OK thanks alot. Will look into it
